Feature/mc 15688 metrics datasource docs (#422)
cloudops/cloudmc-api-docs: source/includes/administration/_metrics.md

`endDate`<br/>*Date* | The start date of the interval to query metrics for. The date must be in a valid ISO-8601 format.  The end date must be provided to query this endpoint.

Optional Query Parameters | &nbsp;
---- | -----------
`environment`<br/>*String* | The name of the environment to query metrics for.
`aggregationType`<br/>*String* | How the queried metrics should be aggregated. The possible values are sum, count, min, max, avg (average). When not provided, the default value used is `average`.
`size`<br/>*Int* | The size of the granularity used to aggregate data points. The size must be a positive integer value. When combined with `unit`, this forms an expression similar to `1 hour`, or `1 day`, etc. When not provided, the default value used is `5`.
`unit`<br/>*String* | The unit of the granularity used to aggregate data points. The unit must be a valid Java Chronounit. When not provided, the default value is `minutes`. 
`entityType`<br/>*String* | the type of entity to query metrics for.
`entityId`<br/>*UUID* | The id of the entity to query metrics for.

> `groupBy`<br/>*String* | .

Author: Not super sure what to say about these 3

Reviewer: For now, we can omit it. When we work on https://cloudops.atlassian.net/browse/MC-15838 then we can update this section.

Author: e2abc9a59af36b60ed34006eeb6e5c5af08b8b5d

Reviewer: for entityType and entityId, how about something like this maybe, if it make sense - 
```
`entityType`<br/>*String* | the type of entity to query metrics for. When provided, only those metric records that have the provided `entityType` are aggregated.
`entityId`<br/>*UUID* | The id of the entity to query metrics for. When provided, only those metric records that have the provided `entityId` are aggregated.
```

Author: I like that! 587c2070ae00dfa2d03e9bb553aeaf12eb272df8

## Metrics

> The metrics API can be queried to obtain data related to different organizations, environments, or workloads

Reviewer: How about something like 👇!!
```suggestion
The metrics API can be queried to obtain metrics data that is pushed by the service to CloudMC metric data store.
```

Author: 92ddbd66507446ccc3ce96a1d3eef48998a24b34

                "label": "2021-08-14T12:00:00.000",
                "value": 0.30
            },
            {
                "label": "2021-08-15T00:00:00.000",
                "value": 0.27
            }
        ]
    }
}
```

Path Parameters | &nbsp;
---- | -----------
`serviceCode`<br/>*String* | A globally unique code that identifies this service connection..
`metricId`<br/>*String* | A unique identifier used to define a `MetricDescriptor`.

Required Query Parameters | &nbsp;
---- | -----------
`startDate`<br/>*Date* | The start date of the interval to query metrics for. The date must be in a valid ISO-8601 format. The start date must be provided to query this endpoint.
`endDate`<br/>*Date* | The start date of the interval to query metrics for. The date must be in a valid ISO-8601 format.  The end date must be provided to query this endpoint.

Optional Query Parameters | &nbsp;
---- | -----------
`environment`<br/>*String* | The name of the environment to query metrics for.
`aggregationType`<br/>*String* | How the queried metrics should be aggregated. The possible values are sum, count, min, max, avg (average). When not provided, the default value used is `average`.

> `size`<br/>*Int* | The size of the granularity used to aggregate data points. The size must be a positive integer value. When combined with `unit`, this forms an expression similar to `1 hour`, or `1 day`, etc. When not provided, the default value used is `5`.

Reviewer: How about 👇 !!
```suggestion
`size`<br/>*Int* | The size of the data points granularity in response. The size must be a positive integer value. When combined with `unit`, this forms an expression similar to `1 hour`, or `1 day`, etc. The default value used is `5`.
```

Author: 92ddbd66507446ccc3ce96a1d3eef48998a24b34

> `unit`<br/>*String* | The unit of the granularity used to aggregate data points. The unit must be a valid Java Chronounit. When not provided, the default value is `minutes`. 

Reviewer: similarly -
```suggestion
`unit`<br/>*String* | The unit of the data points granularity in response. The unit must be a valid Java Chronounit. The default value is `MINUTES`. 
```

Author: 92ddbd66507446ccc3ce96a1d3eef48998a24b34

`entityType`<br/>*String* | the type of entity to query metrics for.
`entityId`<br/>*UUID* | The id of the entity to query metrics for.
`groupBy`<br/>*String* | .

Attributes | &nbsp;
---- | -----------

> `label`<br/>*UUID* | The timestamp used to label the value returned by the query.

Reviewer: Should we mention that the timestamp is in UTC?? 🤔

Author: 92ddbd66507446ccc3ce96a1d3eef48998a24b34

> `value`<br/>*Object* | The value of the metric at the given timestamp.

Reviewer: there is a third attribute as well as entityId if I remember correctly. Can we check pls?

Reviewer: also the type for `value` is `Number` and `label` is `string`. Can we also update those please?

Author: I don't think so, this is what I get as a response when making a query:

```{
    "data": {
        "ingress_bandwidth": [
            {
                "label": "2021-08-12T12:00:00.000",
                "value": 0.27
            },
            {
                "label": "2021-08-13T00:00:00.000",
                "value": 0.29
            },
            {
                "label": "2021-08-13T12:00:00.000",
                "value": 0.25
            },
            {
                "label": "2021-08-14T00:00:00.000",
                "value": 0.37
            },
            {
                "label": "2021-08-14T12:00:00.000",
                "value": 0.26
            },
            {
                "label": "2021-08-15T00:00:00.000",
                "value": 0.21
            }
        ]
    }
}```

Reviewer: Interesting. That is because you are not passing any siteId as a query param. Use `entityId` as the query param key and try passing `siteId`.
As of now the data that we have here is not tied to any particular site but queried for all the sites.

Author: 9a51b5ac02e32fbdf5276033672a44888884ce5f

## Metrics

> The metrics API can be queried to obtain metrics data rthat is pushed by the service to CloudMC metric data store.

Reviewer: a small typo
```suggestion
The metrics API can be queried to obtain metrics data that is pushed by the service to CloudMC metric data store.
```

Author: 5f5a10970c1c8ffb5a9837eb8de7a88b2f193459

Required Query Parameters | &nbsp;
---- | -----------
`startDate`<br/>*Date* | The start date of the interval to query metrics for. The date must be in a valid ISO-8601 format. The start date must be provided to query this endpoint.
`endDate`<br/>*Date* | The start date of the interval to query metrics for. The date must be in a valid ISO-8601 format.  The end date must be provided to query this endpoint.

Optional Query Parameters | &nbsp;
---- | -----------
`environment`<br/>*string* | The name of the environment to query metrics for.
`aggregationType`<br/>*string* | How the queried metrics should be aggregated. The possible values are sum, count, min, max, avg (average). When not provided, the default value used is `average`.
`size`<br/>*integer* | The size of the data points granularity in the response. The size must be a positive integer value. When combined with `unit`, this forms an expression similar to `1 hour`, or `1 day`, etc. The default value used is `5`.
`unit`<br/>*string* | The unit of the data points granularity in the response. The unit must be a valid Java Chronounit. The default value is `MINUTES`.

> `entityType`<br/>*String* | the type of entity to query metrics for. When provided, only those metric records that have the provided `entityType` are aggregated.

Reviewer: Another one 😱 
```suggestion
`entityType`<br/>*String* | The type of entity to query metrics for. When provided, only those metric records that have the provided `entityType` are aggregated.
```

Author: 5f5a10970c1c8ffb5a9837eb8de7a88b2f193459
